pkg/ingestor/parser/spdx/parse_spdx_test.go

```diff
@@ -517,7 +517,7 @@ func Test_spdxParser(t *testing.T) {
 					{
 						Pkg:             pUrlToPkgDiscardError("pkg:guac/spdx/testsbom"),
 						DepPkg:          pUrlToPkgDiscardError("pkg:guac/files/sha1:ba1c68d88439599dcca7594d610030a19eda4f63?filename=./include-file"),
-						DepPkgMatchFlag: generated.MatchFlags{Pkg: generated.PkgMatchTypeAllVersions},
+						DepPkgMatchFlag: generated.MatchFlags{Pkg: generated.PkgMatchTypeSpecificVersion},
```

Comment:
ah... i think in this case, it is set to `AllVersions` since we do not actually version files... 

see my comment in https://docs.google.com/document/d/11WqkncYYob8MtNkcvTZiYcjbvclT15UKFh6coDjJToU/edit, on "At the same time, propose changing files to have the similar format of: `guac/generic/file/sha256@…`, so that they can be referenced via their specific version."

Reply:
I will work to change the ways the files purls are created and handled.

Reply:
check how the file purl is being parsed.

Reply:
digest should not be part of the version to ensure any star graphs.

Reply:
@lumjjb just to clarify here the digest should not be part of the version? We would need to for isDep to work only at the pkgVersion level?

Reply:
i think based on our discussion, there will never be a version here, BUT it should be ok since it defaults to no version being the empty version, so to work towards not using "AllVersions" where possible, this is fine.

Reply:
Ok. The current behavior is that `files` do not have a version (which will result in an empty version string). So this is all good.

Reply:
For example: `"pkg:guac/files/sha1:ba1c68d88439599dcca7594d610030a19eda4f63?filename=./include-file"` 

will create:

Type - `guac`
Namespace - `files`
name - `sha1:ba1c68d88439599dcca7594d610030a19eda4f63?`
version - `""`
qualifier - `"filename"` = `"./include-file"`
subpath - `""`